File: PS Multi Tools/Classes/DownloadQueueItem.vb
Imports System.ComponentModel
Imports System.Runtime.CompilerServices

Public Class DownloadQueueItem
    Implements INotifyPropertyChanged

    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

    Protected Sub NotifyPropertyChanged(<CallerMemberName> Optional propertyName As String = "")
        RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
    End Sub

    Private _GameID As String
    Private _FileName As String
    Private _PKGSize As String
    Private _DownloadURL As String
    Private _MergeState As String
    Private _DownloadState As String

    Public Property GameID As String
        Get
            Return _GameID
        End Get
        Set
            _GameID = Value
            NotifyPropertyChanged("GameID")
        End Set
    End Property

    Public Property FileName As String
        Get
            Return _FileName
        End Get
        Set
            _FileName = Value
            NotifyPropertyChanged("FileName")
        End Set
    End Property

    Public Property PKGSize As String
        Get
            Return _PKGSize
        End Get
        Set
            _PKGSize = Value
            NotifyPropertyChanged("PKGSize")
        End Set
    End Property

    Public Property DownloadURL As String
        Get
            Return _DownloadURL
        End Get
        Set
            _DownloadURL = Value
            NotifyPropertyChanged("DownloadURL")
        End Set
    End Property

    Public Property DownloadState As String
        Get
            Return _DownloadState
        End Get
        Set
            _DownloadState = Value
            NotifyPropertyChanged("DownloadState")
        End Set
    End Property

    Public Property MergeState As String
        Get
            Return _MergeState
        End Get
        Set
            _MergeState = Value
            NotifyPropertyChanged("MergeState")
        End Set
    End Property

End Class

File: PSMultiTools/Classes/DownloadQueueItem.cs
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace PSMultiTools.Classes
{

    public class DownloadQueueItem : INotifyPropertyChanged
    {

        private string? _GameID;
        private string? _FileName;
        private string? _PKGSize;
        private string? _DownloadURL;
        private string? _MergeState;
        private string? _DownloadState;

        public string? GameID
        {
            get => _GameID;
            set => SetProperty(ref _GameID, value);
        }

        public string? FileName
        {
            get => _FileName;
            set => SetProperty(ref _FileName, value);
        }

        public string? PKGSize
        {
            get => _PKGSize;
            set => SetProperty(ref _PKGSize, value);
        }

        public string? DownloadURL
        {
            get => _DownloadURL;
            set => SetProperty(ref _DownloadURL, value);
        }

        public string? DownloadState
        {
            get => _DownloadState;
            set => SetProperty(ref _DownloadState, value);
        }

        public string? MergeState
        {
            get => _MergeState;
            set => SetProperty(ref _MergeState, value);
        }

        public event PropertyChangedEventHandler? PropertyChanged;

        protected bool SetProperty<T>(ref T field, T value, [CallerMemberName] string? propertyName = null)
        {
            if (EqualityComparer<T>.Default.Equals(field, value))
                return false;

            field = value;
            OnPropertyChanged(propertyName);
            return true;
        }

        protected void OnPropertyChanged([CallerMemberName] string? propertyName = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

    }
}